### DISCUSSION TITLE
PTS200 ESP32 based iron possible ironos port

### DISCUSSION THREAD

**Discussion by pykpkg47** [Ralim/IronOS#1771] [Q&A]
the PTS200  is a esp32s2 based iron  sold by some chinese shops.  it looks like the github user [[Edddddddd](https://github.com/Eddddddddy/Songguo-PTS200/commits?author=Eddddddddy)](https://github.com/Eddddddddy) is behind this device. there exist a public repo with the source code of the firmeware(sadly license file missing so no true open source firmeware).

Is it possible to port ironos from stm32/riscv  to esp32.

pinout: https://templates.blakadder.com/PTS200.html

**Comment by Ralim** [comment]
Hia,
Absolutely this _could_ be supported. 

It would require a fair amount of hacking to make the esp-idf framework want to work in the current build system but it could be done. ESP32 already supports FreeRTOS so there is no porting there to be done, would be a fairly simple port apart from figuring out a build system setup for it.

Only problem I can see is the dumb USB-PD controller wont let IronOS read out the current limit; so it would be possible to overload power adaptors.

At this point in time i dont have a huge amount of time to do a new port; but if there is sufficent interest I can look at picking one up.

**Reply by bam80** [reply]
Here is another ESP32-IDF/RTOS based firmware, maybe we could borrow the build system setup or whatever we need from it:
https://github.com/vortigont/ESPIron-PTS200/

**Comment by uejji** [comment]
I notice some vendors are selling a "PTS200 v2" but I can't find any information online about whether this is actually a different hardware revision to the soldering iron.

**Comment by bam80** [comment]
Hi, any news on this? The port is very anticipated.

**Reply by Ralim** [reply]
No one with hardware in-hand has elected to work on this port as far as I'm aware.

**Reply by bam80** [reply]
Could we still do the porting? I'm going to purchase this device, so could assist in testing, etc.

**Reply by Ralim** [reply]
I _can't_ at this point in time, I just can't commit the time.
Anyone in the community is welcome to lead the effort and I can help, but I cant lead the effort.
